### PR TITLE
Remove adapter lines entirely from samplesheet

### PR DESCRIPTION
**Pull request by grst**
Suggesting to remove the adapter lines entirely from the samplesheet when `remove_adapters` is set to true. 
This should make bcl2fastq samplesheets compatible with bclconvert. 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/demultiplex/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/demultiplex _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by grst** [comment]
@nf-core-bot fix linting

**Comment by grst** [comment]
currently doesn't work on S3

**Comment by grst** [comment]
> currently doesn't work on S3

never mind, it was just an encoding issue in the samplesheet

File: workflows/demultiplex.nf [hunk 1]
```diff
@@ -70,16 +70,12 @@ workflow DEMULTIPLEX {
             samplesheet
                 .readLines()
                 .each { line ->
-                    if ( line =~ /Adapter,[ACGT]+,/ ) {
-                        new_line = line.replaceAll(/Adapter,[ACGT]+,/, 'Adapter,,')
-                        removal_checker = true
-                    } else if ( line =~ /AdapterRead2,[ACGT]+,/ ) {
-                        new_line = line.replaceAll(/AdapterRead2,[ACGT]+,/, 'AdapterRead2,,')
+                    if ( line =~ /Adapter.*,[ACGT]+,/ ) {
```
Comment (grst) [inline workflows/demultiplex.nf:73]: Matches Adapter, AdapterRead1, AdapterRead2

**Review by apeltzer — APPROVED** [review]

**Comment by grst** [comment]
This will fall apart as soon as we have more than one flowcells with the same samplesheet filenames. This creates a file in the current directory and potentially overwrite each other. I think the correct pattern to use here would be [collectFile](https://www.nextflow.io/docs/latest/operator.html#collectfile). 

https://github.com/nf-core/demultiplex/blob/48bd76913a0b36c0028fe9f460a59f9418a72eb2/workflows/demultiplex.nf#L63

@nschcolnicov, any chance you could look into that?

**Comment by nschcolnicov** [comment]
> This will fall apart as soon as we have more than one flowcells with the same samplesheet filenames. This creates a file in the current directory and potentially overwrite each other. I think the correct pattern to use here would be [collectFile](https://www.nextflow.io/docs/latest/operator.html#collectfile).
> 
> https://github.com/nf-core/demultiplex/blob/48bd76913a0b36c0028fe9f460a59f9418a72eb2/workflows/demultiplex.nf#L63
> 
> @nschcolnicov, any chance you could look into that?

Absolutely, let me address it in this PR.

**Comment by apeltzer** [comment]
Is this ready for review? Looks ready to me :)

**Comment by grst** [comment]
I think this looks good, thanks @nschcolnicov!
Could we have a test case that ensures that the lines are actually removed? Speaking of experience from an internal pipeline where we thought it would remove those lines but didn't for at least three years.

**Comment by nschcolnicov** [comment]
> Is this ready for review? Looks ready to me :)

Yes, thank you!

**Comment by nschcolnicov** [comment]
> I think this looks good, thanks @nschcolnicov! Could we have a test case that ensures that the lines are actually removed? Speaking of experience from an internal pipeline where we thought it would remove those lines but didn't for at least three years.

Absolutely, I'll add one

**Comment by nschcolnicov** [comment]
@apeltzer @grst Added an additional profile for testing the samplesheet since the only one we had with adapters is test_full. Also, since I was already working on creating a short paired end profile for CI (https://github.com/nf-core/demultiplex/issues/189), I addressed both issues in this PR.

**Review by nschcolnicov — APPROVED** [review]

**Review by apeltzer — APPROVED** [review]

**Pull request merged**